src/main/java/com/monkeyrizz/codeforest/controller/SolutionController.java

```diff
@@ -0,0 +1,29 @@
+package com.monkeyrizz.codeforest.controller;
+
+import com.monkeyrizz.codeforest.model.Solution;
+import com.monkeyrizz.codeforest.service.SolutionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "solution")
+public class SolutionController {
+    private final SolutionService solutionService;
+
+    @Autowired
+    public SolutionController(SolutionService solutionService) {
+        this.solutionService = solutionService;
+    }
```

Comment:
Am I missing something here? AFAIK, `@Autowired` should inject SolutionService into the SolutionController automatically if you annotate solutionService with `@Autowired` -- it is bad practice to have to use the "new" keyword every time we want to create a new solution service and pass it into the solution controller.

e.g.
```
@Autowired
private final SolutionService solutionService;
```

Reply:
https://springframework.guru/best-practices-for-dependency-injection-with-spring/

they put autowire next to the function